src/Wizard.php

```diff
@@ -189,4 +189,63 @@ public function all(): array
         return $this->steps;
     }
 
+    public function replaceStep(int $index, string $new_step_class, string $key): Step
+    {
+        $step = $this->steps[$index];
+        $step->clearData();
+
+        $this->steps[$index] = $this->createStepClass($new_step_class, $index + 1, $key, $index);
+
+        return $this->steps[$index];
+    }
+
+    public function appendStep(string $step_class, string $key): Step
+    {
+        $new_index = count($this->steps);
+        $this->steps[] = $this->createStepClass($step_class, $new_index + 1, $key, $new_index);
+
+        return $this->steps[$new_index];
+    }
+
+    public function insertStep(int $index, string $step_class, string $key): Step
+    {
+        if ($index >= count($this->steps) || count($this->steps) === 0) {
```

Comment:
Hey, it looks good :)
I see here way to make a little tweak/upgrade here. On most pessimistic path we count property `$this->steps` 3 times lets move this to new property `$steps_count`